src/preload.ts

```diff
@@ -106,6 +106,24 @@ const statIsDirectory = (path: string) =>
   stat(path).then((res) => res.isDirectory())
 const getPath = async (name: string) => ipcRenderer.invoke('app.getPath', name)
 
+const canReadWriteDirectory = async (path: string) : Promise<{value: boolean, error: unknown }> => {
+  const isDirectory = await statIsDirectory(path)
+  if (!isDirectory) {
+    throw new Error('path is not a directory. Do not send a file path.')
+  }
+
+  // bitwise OR to check read and write permissions
+  try {
+    const canReadWrite = await fs.access(path, fs.constants.R_OK | fs.constants.W_OK)
```

Comment:
https://nodejs.org/api/fs.html#fsaccesspath-mode-callback

> Do not use fs.access() to check for the accessibility of a file before calling fs.open(), fs.readFile(), or fs.writeFile(). Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file is not accessible.

I am specifically make this a directory check. I do not do any `fs.access` on files. This should be safe for directories.